tedana/model/fit.py

```diff
@@ -8,90 +8,87 @@
 import pandas as pd
 from scipy import stats
 
-from tedana import (combine, io, utils)
+from tedana import io, utils
 
 LGR = logging.getLogger(__name__)
 
 F_MAX = 500
 Z_MAX = 8
 
 
-def fitmodels_direct(catd, mmix, mask, t2s, t2s_full, tes, combmode, ref_img,
-                     reindex=False, mmixN=None, full_sel=True, label=None,
-                     out_dir='.', verbose=False):
+def dependence_metrics(catd, tsoc, mmix, mask, t2s, tes, ref_img,
+                       reindex=False, mmixN=None, method=None, label=None,
+                       out_dir='.', verbose=False):
     """
     Fit TE-dependence and -independence models to components.
 
     Parameters
     ----------
     catd : (S x E x T) array_like
         Input data, where `S` is samples, `E` is echos, and `T` is time
+    tsoc : (S x T) array_like
+        Optimally combined data
     mmix : (T x C) array_like
         Mixing matrix for converting input data to component space, where `C`
         is components and `T` is the same as in `catd`
     mask : (S [x E]) array_like
         Boolean mask array
     t2s : (S [x T]) array_like
         Limited T2* map or timeseries.
-    t2s_full : (S [x T]) array_like
-        Full T2* map or timeseries. For voxels with good signal in only one
-        echo, which are zeros in the limited T2* map, this map uses the T2*
-        estimate using the first two echoes.
     tes : list
         List of echo times associated with `catd`, in milliseconds
-    combmode : {'t2s', 'paid'} str
-        How optimal combination of echos should be made, where 't2s' indicates
-        using the method of Posse 1999 and 'paid' indicates using the method of
-        Poser 2006
     ref_img : str or img_like
         Reference image to dictate how outputs are saved to disk
     reindex : bool, optional
-        Default: False
-    mmixN : array_like, optional
-        Default: None
-    full_sel : bool, optional
-        Whether to perform selection of components based on Rho/Kappa scores.
-        Default: True
+        Whether to sort components in descending order by Kappa. Default: False
+    mmixN : (T x C) array_like, optional
+        Z-scored mixing matrix. Default: None
+    method : {'kundu_v2', 'kundu_v3', None}, optional
```

Comment:
Maybe we could call this `decision` ?

Reply:
Does `None` make sense as the default argument ? What behavior do we expect when we return an empty `seldict` ?

Reply:
I'm also still really unsure that we should have `kundu_v3` here, since it's not reintegrated, and we don't know when it will be.

Reply:
`None` is the PCA decision tree-compatible version, which doesn't return metric maps and only gives back the component table.

`kundu_v3` requires certain metric maps in `seldict` (e.g., `PSC`, `WTS`, and `tsoc_B`) that `kundu_v2` does not. I didn't want to remove those metric maps from the overall function, because it would be annoying to re-implement them later, but also didn't want to output an oversized version of `seldict` when it's not necessary. I can drop it from the arguments and docstring, but I'd like to keep it in the if/else statement below.

Reply:
I don't think `decision` is right, since that implies that actual decisions are made in this function. It really just determines which outputs will be generated. PCA requires just the component table, while Kundu ICA v2 requires that and several metric maps, and Kundu ICA v3 requires the component table and even more metric maps.

What about `algorithm`?